src/objects/zcl_abapgit_objects.clas.abap

```diff
@@ -513,21 +516,32 @@ CLASS zcl_abapgit_objects IMPLEMENTATION.
               iv_package = <ls_tadir>-devclass
               is_item    = ls_item ).
 
+            INSERT <ls_tadir> INTO TABLE lt_deleted.
             DELETE lt_tadir.
             lv_count = lv_count + 1.
 
             " make sure to save object deletions
             COMMIT WORK.
-          CATCH zcx_abapgit_exception INTO lx_error ##NO_HANDLER.
-            " ignore errors inside the loops and raise it later
+          CATCH zcx_abapgit_exception INTO lx_error.
+            IF ii_log IS BOUND.
+              ii_log->add_exception( ix_exc  = lx_error
+                                     is_item = ls_item ).
+              ii_log->add_error( iv_msg = |Deletion of object { ls_item-obj_name } failed|
+                                 is_item = ls_item ).
+            ENDIF.
         ENDTRY.
 
       ENDLOOP.
+
+      " Exit if done or nothing else was deleted
+      IF lines( lt_tadir ) = 0 OR lines( lt_deleted ) = 0.
+        EXIT.
+      ENDIF.
     ENDDO.
 
     zcl_abapgit_default_transport=>get_instance( )->reset( ).
 
-    IF lx_error IS BOUND AND lines( lt_tadir ) > 0.
+    IF lx_error IS BOUND AND lines( lt_tadir ) > 0 AND NOT ii_log IS BOUND.
```

Comment:
Without raising the exception in the case of using the log the uninstall process does not abort on error. So you get the error popup but when you close it the repo is still removed even though objects still exist.

Reply:
back to showing the last error. or should we make it a generic one "Error during uninstall. Check the log"?

after it showed the popup, did it show the "Log" menu item, so you could go back to it?

Reply:
Not sure.

After I closed the automatically opened log popup I was immediately redirected to the repository overview where the repo wasn't listed anymore.

Reply:
I think now it's mostly fine. You get the error at the bottom (a bit generic, yes, but probably caused by the error handling in the specific object type I guess), you get the multiple errors at the top and can also open the log for callstack / longtexts.

![image](https://user-images.githubusercontent.com/5270359/99538960-f582c080-29ad-11eb-80f7-818a8b7ad88f.png)